script.sh

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash
+
+
+log_file=""
+method=1
+country_flag=false
+
+# Help 
+show_help() {
+  echo "Usage: $0 [-f <file>] [-m <method>] [-c] [-h]"
+  echo ""
+  echo "Options:"
+  echo "  -f, --file     Path to the Nginx log file"
+  echo "  -m, --method   Parsing method (1: awk, 2: grep)"
+  echo "  -c, --country  Fetch country information for each IP"
+  echo "  -h, --help     Show help information"
+  echo ""
+  echo "Examples:"
+  echo "  $0 -f nginx.log -m 1"
+  echo "  $0 -f nginx.log -m 2 -c"
+}
+
+# Перший метод через awk
+parse_with_awk() {
+  echo "Using awk to parse the log file..."
+  awk '{print $1}' "$log_file" | sort | uniq -c | sort -nr | awk '{print $2, $1}'
+}
+
+# Другий метод через grep
+parse_with_grep_cut() {
+  echo "Using grep to parse the log file..."
+  grep -oP '^\d+\.\d+\.\d+\.\d+' "$log_file" | sort | uniq -c | sort -nr | awk '{print $2, $1}'
+}
+
+# Функція визначення країни
+get_country() {
+  ip=$1
+  country=$(curl -s https://ipinfo.io/$ip | jq -r '.country')
+  echo "$country"
+}
+
+#Додавання столбця країни до видачі 
+add_country_info() {
```

Comment:
Your method tries to get the country by private IP, and it will be null. I think it would be nice to have a filter without private IPs when someone wants to get the country IP address 
```
❯ ./script.sh -f nginx.log -m 1 -c
Using awk null
127.0.0.1 4354 null
172.31.91.218 723 null
172.31.25.146 723 null
```